app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -383,13 +383,16 @@ def save_pulp_push_repository_href
           :not_found
         )
       end
+
+      enable_container_push_repo_update(root_repository)
       instance_repo.update!(version_href: latest_version_href)
       # The Pulp repository should not change after first creation
       if root_repository.repository_references.empty?
         ::Katello::Pulp3::RepositoryReference.where(root_repository_id: instance_repo.root_id,
                                                     content_view_id: instance_repo.content_view.id,
                                                     repository_href: pulp_repo_href).create!
       end
+      disable_container_push_repo_update(root_repository)
```

Comment:
Especially since `update!` raises errors if anything is invalid, what do you think about putting the `disable_container_push_repo_update` line inside an `ensure` block? This way if that happens, you won't be left with an incorrectly-update-enabled container push repo.

Reply:
Thank you for the input with this, Jeremy. I was able to work around the lock entirely with the latest push so I'm marking this as resolved.